src/utils/config_utils/src/config_utils.c

```diff
@@ -57,13 +57,64 @@ static const char* CONFIG_RUN_AS = "runas";
 static const char* CONFIG_CONNECTION_SOURCE = "connectionSource";
 static const char* CONFIG_CONNECTION_TYPE = "connectionType";
 static const char* CONFIG_CONNECTION_DATA = "connectionData";
+static const char* CONFIG_CONNECTION_X509_CERT_FILE_PATH = "connectionX509CertFilePath";
+static const char* CONFIG_CONNECTION_X509_PRIVATE_KEY_FILE_PATH = "connectionX509PrivateKeyFilePath";
+static const char* CONFIG_CONNECTION_X509_CA_CERT_FILE_PATH = "connectionX509CaCertFilePath";
 static const char* CONFIG_ADDITIONAL_DEVICE_PROPERTIES = "additionalDeviceProperties";
 static const char* CONFIG_AGENTS = "agents";
 
 static const char* INVALID_OR_MISSING_FIELD_ERROR_FMT = "Invalid json - '%s' missing or incorrect";
 
 static void ADUC_AgentInfo_Free(ADUC_AgentInfo* agent);
 
+
+static char* ADUC_AgentInfo_Read_X509_File(const char* const x509_file_path) {
+    char* buffer = NULL;
+    size_t bufferLength = 0;
+    size_t bytesRead = 0;
+    FILE* x509File = NULL;
+    
+    x509File = fopen(x509_file_path, "rb");
+    if (x509File)
+    {
+        if (fseek(x509File, 0, SEEK_END) == 0) {
+            bufferLength = (size_t)ftell(x509File);
+            if (fseek(x509File, 0, SEEK_SET) == 0)
+            {
+                buffer = malloc(bufferLength);
+                if (buffer)
+                {
+                    bytesRead = fread(buffer, 1, bufferLength, x509File);
+                    if (bytesRead < bufferLength)
+                    {
+                        Log_Error("Failed to read '%s'!", x509_file_path);
```

Comment:
could eof and error be distinguished and also log the return value from bytesRead? Something like this:
```c
if (bytesRead < bufferLength)
{
    if (feof(x509File)) {
        Log_Error("Unexpected end of file while reading '%s', bytesRead: %zu", x509_file_path, bytesRead);
    } else if (ferror(x509File)) {
        Log_Error("Error occurred while reading '%s', bytesRead: %zu", x509_file_path, bytesRead);
    }
    free(buffer);
    buffer = NULL;
}
```